.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,193 @@
+
+Otherwise, your feature may not be approved at all.
+
+### Building a new feature
+
+To build a new feature, check out a new branch based on the `master` branch, and be sure to consider the following:
+
+- If the feature has a public facing API, make sure to document it, using code comments
+
+</details>
+
+<br>
+
+<details>
+<summary>Contribute to the documentation :black_nib:</summary>
+<br>
+
+The C client library documentation is hosted on https://docs.iota.org, which is built from content in the [documentation](https://github.com/iotaledger/documentation) repository.
+
+Please see the [guidelines](https://github.com/iotaledger/documentation/CONTRIBUTING.md) on the documentation repository for information on how to contribute to the documentation.
+</details>
+
+<br>
+
+<details>
+<summary>Pull requests :mega:</summary>
+<br>
+
+This section guides you through submitting a pull request (PR). Following these guidelines helps give your PR the best chance of being approved and merged.
+
+### Before submitting a pull request
+
+Before submitting a pull request, do the following:
+
+- Install Git hooks by running `./tools/hooks/autohook.sh install` from the `iota.c` directory
+
+    These Git hooks allow you to commit code only after it passes the formatting checks.
+
+- Install the following requirements for the Git hooks 
+
+    - [Buildifier](https://github.com/bazelbuild/buildtools/tree/master/buildifier) 
+
+    - [ClangFormat](https://clang.llvm.org/docs/ClangFormat.html) 
+
+    Buildifier can be installed with Bazel or Go.  
+
+    To install Buildifier with Bazel:
+
+    1. Clone `bazelbuild/buildtools` repository
+   `$ git clone https://github.com/bazelbuild/buildtools.git`
+
+    2. Change into the `buildtools` directory
+
+    3. build it with bazel command, `$ bazel build //buildifier`
+    The executable file will be located under `path/to/buildtools/bazel-bin`
+
+    4. create a soft link
+
+    To install Buildifier with Go:
+
+    1. Change into the `$GOPATH` directory
+
+    2. Run `$ go get github.com/bazelbuild/buildtools/buildifier`
+    The executable file will be located under `$GOPATH/bin`
+
+    3. create a soft link for global usage, run
+    `$ sudo ln -s $HOME/go/bin/buildifier /usr/bin/buildifier`
+
+    To install ClangFormat on a Debian/Ubuntu-based operating system: `$ sudo apt-get install clang-format`
+    To install ClangFormat on macOS operating system:`$ brew install clang-format`
+
+When creating a pull request, please follow these steps to have your contribution considered by the maintainers:
+
+- A pull request should have only one concern (for example one feature or one bug). If a PR addresses more than one concern, it should be split into two or more PRs.
+
+- A pull request can be merged only if it references an open issue
+
+    **Note:** Minor changes such as fixing a typo can but do not need an open issue.
+
+- All code should be well tested
```

Comment:
The running environment of iota.c could be vary, the functionality can be affected by different hardware architectures, compilers, or operating systems. it's hard to define a `well tested` code or we should have a standard in README or somewhere.
```suggestion
- All code should be covered in unit tests 
```

Reply:
Do we mean that code should pass the unit tests (are this in the Git hooks?), or that you should write unit tests for your code?

Reply:
there are unit tests, like `iota.c/cclient/api/tests`, those will be tested through CI. The contributor might need to add new tests for new functionality. I think it is mentioned in the PR section: https://github.com/iotaledger/iota.c/pull/23/files#diff-8cc7b2b0d78dd2501610391c086a8516R184